lib/mysql2/aurora.rb

```diff
@@ -29,22 +29,22 @@ def query(*args)
         begin
           client.query(*args)
         rescue Mysql2::Error => e
+          raise e unless e.message&.include?('--read-only')
```

Comment:
You may prefer the previous `if` with single `raise`.  I do prefer guard conditions when possible but it's your call.